lib/bundler/environment_preserver.rb

```diff
@@ -1,22 +1,37 @@
 # frozen_string_literal: true
 module Bundler
   class EnvironmentPreserver
+    INTENTIONALLY_NIL = "BUNDLER_ENVIRONMENT_PRESERVER_INTENTIONALLY_NIL".freeze
+    BUNDLER_KEYS = %w[
```

Comment:
It's a small enough array that linearly searching it will probably be faster than the indirection in a set